editor/plugins/material_editor_plugin.cpp

```diff
@@ -95,11 +103,20 @@ void MaterialEditor::_notification(int p_what) {
 	}
 }
 
+void MaterialEditor::_set_rotation(real_t p_x_degrees, real_t p_y_degrees) {
+	rot.x = Math::deg_to_rad(p_x_degrees);
+	rot.y = Math::deg_to_rad(p_y_degrees);
+	_update_rotation();
+}
+
 void MaterialEditor::_update_rotation() {
 	Transform3D t;
 	t.basis.rotate(Vector3(0, 1, 0), -rot.y);
 	t.basis.rotate(Vector3(1, 0, 0), -rot.x);
 	rotation->set_transform(t);
+	// Store the rotation so it can persist when switching between materials.
+	Vector2 rotation_degrees = Vector2(Math::rad_to_deg(rot.x), Math::rad_to_deg(rot.y));
+	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_rotation", rotation_degrees);
```

Comment:
This shouldn't be here. It should happen only when rotating (in `gui_input()`) and maybe after changing shape. Otherwise you store it more often than needed, including when initializing editor.

Reply:
Yeah, I had it in `gui_input()` but moved it here so that it would be called when the shape was set as well. I can inline it into all those places instead,